adagrams/game.py

```diff
@@ -1,13 +1,129 @@
 from random import randint
-
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
 def draw_letters():
-    pass
+    # turns the dictionary LETTER_POOL into a letters list 
+    letter_deck = []
+    letter_bank = []
+    for letter, freq in LETTER_POOL.items():
+        for i in range(freq):
+            letter_deck.append(letter)
+    
+    for i in range(10):
+        index = randint(0,len(letter_deck)-1)
+        letter_bank.append(letter_deck[index])
+        letter_deck.pop(index)
+
+    return letter_bank
+ 
 
 def uses_available_letters(word, letter_bank):
-    pass
+    filtered = []
+    letter_bank_count = {}
+    # Count letters in the bank, store in a dict
+    for letter in letter_bank:
+        letter_bank_count[letter] = letter_bank_count.get(letter,0) + 1
+
+
+
+    for letter in list(word.upper()):
+        if letter_bank_count.get(letter,0) > 0:
+            filtered.append(letter)  # Build filtered list and check availability
+            letter_bank_count[letter] -= 1 # decrease count in dict only
+        else:
+            return False
+
+   # Check if all letters were used
+    if filtered == list(word.upper()):
+        return True
+    else:
+        return False
+
+# rethinking on 9/19/2025 of the function uses_available_letters:
+# changes and reason :
+# I was building a separate filtered list. But I removed it bc the loop already checks if each letter is available in the letter bank, so if the loop finishes, we know the word can be formed.
+# updated function :
+# def uses_available_letters(word, letter_bank): 
+#     letter_bank_count = {} 
+#     for letter in letter_bank: 
+#         letter_bank_count[letter] = letter_bank_count.get(letter, 0) + 1 
+        
+#     for letter in word.upper(): 
+#         if letter_bank_count.get(letter, 0) == 0: 
+#             return False 
+#         letter_bank_count[letter] -= 1 
+        
+#     return True
+
+    
 
 def score_word(word):
-    pass
+    LETTER_SCORES = { "A": 1, "E": 1, "I": 1, "O": 1, "U": 1, "L": 1, "N": 1, "R": 1, "S": 1, "T": 1, 
+                     "D": 2, "G": 2, "B": 3, 
+                     "C": 3, "M": 3, "P": 3, 
+                     "F": 4, "H": 4, "V": 4, "W": 4, "Y": 4, 
+                     "K": 5, 
+                     "J": 8, "X": 8, 
+                     "Q": 10, "Z": 10 
+    }
+    score = 0
+    if not word:
+        return score
+    else:
+        
+        for letter in word.upper():
+            score += LETTER_SCORES[letter]
+
+    if len(word) >= 7:
+        score += 8
```

Comment:
This is another place where Magic Numbers have appeared! Try to avoid hardcoded values! We could try something like the following to fix it:

```suggestion
        BONUS_SCORE = 8
        score += BONUS_SCORE
```